### PR TITLE
Clean up migrations to work on an empty environment

### LINKED ISSUE #62: Bug: DatoCMS migrations don't run
Trying to run the autogenerated migrations on a new environment throws an error:

```bash
Running migration "1691503643_addTranslations.ts"... !
    Error: Must use import to load ES Module: /path/to/voorhoede/head-start/config/datocms/migrations/1691503643_addTranslations.ts
    require() of ES modules is not supported.
    require() of /Users/jasper/code/voorhoede/head-start/config/datocms/migrations/1691503643_addTranslations.ts from 
    /Users/jasper/code/voorhoede/head-start/node_modules/@datocms/cli/lib/commands/migrations/run.js is an ES module file as it is a .ts file whose nearest parent package.json contains 
    "type": "module" which defines all .ts files in that package scope as ES modules.
    Instead change the requiring code to use import(), or remove "type": "module" from /path/to/voorhoede/head-start/package.json.

    Code: ERR_REQUIRE_ESM
```

The migrations are auto-generated using `npx datocms:new --autogenerate=...` with format set to TypeScript. However our setup seems to cause an issue between what Astro and what DatoCMS expect in terms of `tsconfig.json` and `package.json` > `type`.

### PR DESCRIPTION
**Pull request by Siilwyn**
Closes #62

Chose to add a package.json commonjs boundary to dato so we can keep module in root.
Tested by creating a new dato project and using `datocms migrations:run`.

**After merging we should create a new environment in head start to apply all changes too.**

Possible (future) tasks:
- Simplify migration code, tried some refactoring to e.g. replace hardcoded IDs with model names. Not sure if it's worth it though
- Set up an empty project or environment as a E2E test to run through migrations
- Allow other default locale at the start instead of relying on English
- Delete dato projects?
    - head-start-seeding
    - debug head start migrations

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "typescript": "^5.2.2"
   },
   "devDependencies": {
-    "@datocms/cli": "^1.2.1",
+    "@datocms/cli": "^2.0.5",
```
Comment (Siilwyn) [inline package.json:62]: Updated the CLI, because why not, seems to work.

**Review by decrek — COMMENTED** [review]
In the readme there are warnings about this issue an d the seed script: https://github.com/voorhoede/head-start/blob/migrations-iteration/docs/getting-started.md#create-a-datocms-project. Can you update this section based on your changes?

**Review by decrek — CHANGES_REQUESTED** [review]
Looks really good! Just some documentation updates and I'm happy!

File: docs/getting-started.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Head Start requires Node.js to be installed. See [/.nvmrc](../.nvmrc) for the co
 - Install the project dependencies (`npm install`).
 - Create a `.env` file (`cp .env.example .env`).
 
-```bash
+```shell
```
Comment (Siilwyn) [inline docs/getting-started.md:16]: Replace syntax highlighting language `bash` alias with more appropriate/generic `shell`.

File: docs/getting-started.md [hunk 3]
```diff
@@ -31,14 +31,11 @@ Before you can run your project locally, you need to set-up a DatoCMS project.
...
-> [!WARNING]
-> Head Start has an open [issue on running migrations](https://github.com/voorhoede/head-start/issues/62). For now (instead of creating a blank project and running migrations) duplicate the 'Head Start' project in the Dato dashboard. You can ignore the steps below, but make sure to configure Dato API tokens in your `.env` file and set `datocmsEnvironment = 'main'` in `datocms-environment.ts`.
-
 - [Signup](https://dashboard.datocms.com/signup) or [signin](https://dashboard.datocms.com/) to your DatoCMS account.
-- [Create a new DatoCMS project](https://dashboard.datocms.com/personal-account/projects/browse/new) (select blank project).
+- Create a new blank DatoCMS project.
 - In your CMS, go to Project Settings > API tokens (`/project_settings/access_tokens`) and copy the tokens full-access API token (`DATOCMS_API_TOKEN`) and read-only API token (`DATOCMS_READONLY_API_TOKEN`) to your `.env` file (see below).
 
-```bash
+```dotenv
```
Comment (Siilwyn) [inline docs/getting-started.md:38]: Fix syntax highlighting, this isn't bash.

File: docs/getting-started.md [hunk 4]
```diff
@@ -31,14 +31,11 @@ Before you can run your project locally, you need to set-up a DatoCMS project.
 
 ## Create a DatoCMS project
 
-> [!WARNING]
-> Head Start has an open [issue on running migrations](https://github.com/voorhoede/head-start/issues/62). For now (instead of creating a blank project and running migrations) duplicate the 'Head Start' project in the Dato dashboard. You can ignore the steps below, but make sure to configure Dato API tokens in your `.env` file and set `datocmsEnvironment = 'main'` in `datocms-environment.ts`.
-
 - [Signup](https://dashboard.datocms.com/signup) or [signin](https://dashboard.datocms.com/) to your DatoCMS account.
-- [Create a new DatoCMS project](https://dashboard.datocms.com/personal-account/projects/browse/new) (select blank project).
```
Comment (Siilwyn) [inline docs/getting-started.md:38]: This URL doesn't work for me, simplified the step description a bit.

**Review by jbmoelker — APPROVED** [review]
Works like a charm! Tested it with a blank DatoCMS project. All the migrations run successfully. Schema looks good.

Only suggestions:
- document the `"type": "commonjs"` in `config/datocms/package.json`?
- create a new issue to rebuild head-start datocms instance with these migrations and (manually) copy all the content?

**Pull request merged**